### PR TITLE
[ExportVerilog] Remove temporary vector in text substitution

### PR DESCRIPTION
**Pull request by prithayan**
It is cheaper to call `namify` than to store it in `SmallVector` during text substitution.
Seems to drop runtime of a test case from 150.5509 sec to 2.7712 sec, size of the vector was ~400 elements.

### CODE REVIEW

**Review by seldridge — APPROVED** [review]

**Review by darthscsi — COMMENTED** [review]
can you try storing the stringref in the vector rather than strings?  a good test case is something which substitutes the same thing in a verbatim lots of times ({{1}}{{1}}{{1}}{{1}}{{1}}{{1}}).

File: lib/Conversion/ExportVerilog/ExportVerilog.cpp [hunk 1]
```diff
@@ -687,7 +675,17 @@ void EmitterBase::emitTextWithSubstitutions(
         operandEmitter(op->getOperand(operandNo));
       else if ((operandNo - op->getNumOperands()) < numSymOps) {
         unsigned symOpNum = operandNo - op->getNumOperands();
-        os << symOps[symOpNum];
+        auto sym = symAttrs[symOpNum];
+        SmallString<12> symVerilogName;
```
Comment (lattner) [inline lib/Conversion/ExportVerilog/ExportVerilog.cpp:679]: As Darth says, please change this to be a StringRef, there is no need to copy the stringref into a vector before printing it.

**Review by lattner — APPROVED** [review]

**Comment by prithayan** [comment]
> can you try storing the stringref in the vector rather than strings? a good test case is something which substitutes the same thing in a verbatim lots of times ({{1}}{{1}}{{1}}{{1}}{{1}}{{1}}).

Tried the `SmallVector` of `StringRef`, no improvement in runtime. 
Also, the OMIR test case I was working with, has several such `VerbatimOps`, where the top level module symbol is repeated many many many times, {{0}}{{0}}...

**Comment by prithayan** [comment]
Thanks for the review, merged.

**Comment by darthscsi** [comment]
I see the issue, you are forming a vector the size of the number of substitutions rather than the number of unique symbols.

**Pull request merged**